File: Solutions/349-IntersectionOfTwoArrays/IntersectionOfTwoArrays.ts
function intersection(nums1: number[], nums2: number[]): number[] {
    const set1 = new Set(nums1);
    const set2 = new Set(nums2);
    let result: number[] = [];

    set1.forEach((num) => {
        if (set2.has(num)) result.push(num);
    });

    return result;
}

File: Solutions/349-IntersectionOfTwoArrays/IntersectionOfTwoArrays.go
package solutions

func intersection(nums1 []int, nums2 []int) []int {
	set1 := make(map[int]bool)
	set2 := make(map[int]bool)

	var result []int

	for _, value := range nums1 {
		set1[value] = true
	}

	for _, value := range nums2 {
		set2[value] = true
	}

	for key, _ := range set1 {
		_, exists := set2[key]
		if exists {
			result = append(result, key)
		}
	}

	return result
}
